frontend/src/scenes/surveys/surveysLogic.tsx

```diff
@@ -11,8 +11,23 @@ import { userLogic } from 'scenes/userLogic'
 
 import { AvailableFeature, Breadcrumb, ProgressStatus, Survey, SurveyType } from '~/types'
 
+import { NEW_SURVEY, NewSurvey } from './constants'
 import type { surveysLogicType } from './surveysLogicType'
 
+export function duplicateExistingSurvey(survey: Partial<Survey>): NewSurvey {
+    const copiedNewSurvey = Object.keys(NEW_SURVEY).reduce((acc, key) => {
+        acc[key] = survey[key]
+        return acc
+    }, {}) as NewSurvey
+
+    return {
+        ...copiedNewSurvey,
```

Comment:
not sure why you need to do it like this?

Is the below suggestion not sufficient?

```suggestion
        ...survey,
```

Reply:
reasonable 👍